tests/ui/needless_lifetimes.rs

```diff
@@ -1,3 +1,4 @@
+// edition:2018
```

Comment:
This shouldn't be necessary anymore, since we use the 2021 edition by default for our tests now.

Reply:
Do we? If I remove it I get an error: "Rust 2015 does not permit the use of `async fn`", and the 3 commits that are not in my "master" branch don't seem to change that.

Reply:
Huh? Let me actually try out my suggestion myself. It _should_ work after rebasing.

Reply:
I was out of sync with upstream, I'll rebase and update the test!

Reply:
Thanks! I rebased on the latest upstream master and it worked for me locally.

Reply:
You were right, rebasing allowed me to remove this line.